App.js

```diff
@@ -1,20 +1,43 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+// Components
 import { NavigationContainer } from '@react-navigation/native';
 import { createStackNavigator } from '@react-navigation/stack';
-import HomeScreen from './src/screens/HomeScreen/HomeScreen';
-import SearchScreen from './src/screens/SearchScreen/SearchScreen';
-import LearningScreen from './src/screens/LearningScreen/LearningScreen';
+import Icon from 'react-native-vector-icons/Ionicons';
+import HomeScreen from './src/screens/home/HomeScreen';
+import SearchScreen from './src/screens/search/SearchScreen';
+import LearningScreen from './src/screens/learning/LearningScreen';
+import SettingsScreen from './src/screens/settings/SettingsScreen';
+
+// Context
+import UserContext from './src/contexts/UserContext';
+
+// Utils
+import { getUser } from './src/utils/storage';
+
+Icon.loadFont();
 
 const Stack = createStackNavigator();
 
-const App = () => (
-  <NavigationContainer>
-    <Stack.Navigator>
-      <Stack.Screen name="Home" component={HomeScreen} />
-      <Stack.Screen name="Search" component={SearchScreen} />
-      <Stack.Screen name="Learn" component={LearningScreen} />
-    </Stack.Navigator>
-  </NavigationContainer>
-);
+const App = () => {
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    getUser().then(setUser);
```

Comment:
Does `setUser` need to be a dependency here? For some reason I am thinking that state setters can be omitted from the dependencies since presumably the setter function itself never changes.

Reply:
Yeah this is the obnoxious aspect of how hooks are linted with the some linter defaults — the linter complains if you don't include _every_ dependency.